lib/identifiers/doi.rb

```diff
@@ -1,7 +1,7 @@
 module Identifiers
   class DOI
     def self.extract(str)
-      str.scan(%r{\b10\.\d{3,}/\S+\b}).map(&:downcase)
+      str.scan(%r{\b10.\d{4,9}/[-._;()/:A-Z0-9]+\b}i).map(&:downcase)
```

Comment:
Sadly, this will exclude valid Wiley DOIs which we need to support for Altmetric's use cases.